### DISCUSSION TITLE
Does promptflow support logging LLM calls and promptflow evaluation metrics to azure app insights?

### DISCUSSION THREAD

**Discussion by meghdivya** [microsoft/promptflow#3095] [General]
We want following features:
- Ingesting promptflow logs in app insights 
- Ingesting metrics in app insights (in case of a evaluation) 
- Add custom dimension into the logs and metrics. 
- Log OpenAI / other model call details in app insights: total token counts, prompt token counts, output token counts and latency. 
- Log logprobs details (in case of completion model) and additional responses from OpenAI

**Comment by 0mza987** [comment]
As customers can have really different demands on tracing/logging, we encourage customers to follow OTEL protocols and write their own collectors. Please refer to below guides:

- [promptflow/examples/tutorials/tracing/custom-otlp-collector/otlp-trace-collector.ipynb at main · microsoft/promptflow](https://github.com/microsoft/promptflow/blob/main/examples/tutorials/tracing/custom-otlp-collector/otlp-trace-collector.ipynb)
- [Microsoft OpenTelemetry exporter for Azure Monitor | Microsoft Learn](https://learn.microsoft.com/en-us/python/api/overview/azure/monitor-opentelemetry-exporter-readme?view=azure-python-preview)
